warehouse/query-core/src/main/java/datawave/query/function/IndexOnlyContextCreator.java

```diff
@@ -168,6 +182,18 @@ protected DatawaveJexlContext newDatawaveJexlContext(final Tuple3<Key,Document,M
             newContext = parentContext;
         }
         
+        // see if there are any delayed subtrees that are index only
+        if (delayedIndexOnlySubTreeVisitor != null && delayedIndexOnlySubTreeVisitor.getFoundIndexOnlyFields().size() > 0) {
+            // build the current document range from the document Key to end of the document, even though for some query logics this may be too large a range,
+            // it will be narrowed with equality later
+            Key startKey = new Key(from.first().getRow(), from.first().getColumnFamily());
```

Comment:
Are we creating a document range or a field index range here?

Reply:
IteratorBuildingVisitor expects document ranges